src/error.rs

```diff
@@ -0,0 +1,31 @@
+use std::io;
+
+use reqwest::{self, header::InvalidHeaderValue};
+use thiserror::Error as ThisError;
+
+/// Represents any of the ways that using this library can fail.
+#[derive(ThisError, Debug)]
+pub enum SendgridError {
+    /// The failure was due to some IO error, for example an interrupted network connection.
+    #[error("IO Error: `{0:?}`")]
```

Comment:
With regards to error display, aren't we meant to use `Display` formatters here instead of `Debug` formatters?

Reply:
yeah Display is better

Reply:
I created an issue at: https://github.com/seanmonstar/reqwest/issues/1043, I think it's better to make it addressed in reqwest.